lib/systemd.c

```diff
@@ -20,68 +20,50 @@
  */
 
 #include <zebra.h>
+#include <sys/un.h>
 
 #include "thread.h"
 #include "systemd.h"
 
-#if defined HAVE_SYSTEMD
-#include <systemd/sd-daemon.h>
-#endif
-
-/*
- * Wrapper this silliness if we
- * don't have systemd
+/* these are cleared from env so they don't "leak" into things we fork(),
+ * particularly for watchfrr starting individual daemons
+ *
+ * watchdog_pid is currently not used since watchfrr starts forking.
+ * (TODO: handle that better, somehow?)
  */
+static pid_t watchdog_pid = -1;
+static intmax_t watchdog_usec;
+
+/* not used yet, but can trigger auto-switch to journald logging */
+bool sd_stdout_is_journal;
+bool sd_stderr_is_journal;
+
+static char *notify_socket;
+
+/* talk to whatever entity claims to be systemd ;) */
 static void systemd_send_information(const char *info)
 {
-#if defined HAVE_SYSTEMD
-	sd_notify(0, info);
-#else
-	return;
-#endif
-}
+	int sock;
```

Comment:
IMO this function needs a good doc comment since it's replacing `sd_notify` which does have good docs. It's not obvious what is being done here.

Reply:
I agree, but at the same time I have no idea what to write.  It opens a socket, sends a string, and closes the socket.  The code says that.  Documentation shouldn't duplicate the code itself, and "what?" comments are very rarely appropriate.  But what else is there to document here?

As a sidenote:  the `sd_notify` man page documents the commands and actions you can perform *over* the socket.  What it does is only a single paragraph.

Reply:
A bit of info about what `info` contains might be appropriate but I do see your point

Reply:
```
/* talk to whatever entity claims to be systemd ;)
 *
 * refer to sd_notify docs for messages systemd accepts over this socket.
 * This function should be functionally equivalent to sd_notify().
 */
static void systemd_send_information(const char *info)
```

Pointer to sd_notify should be good?

Reply:
good enough